packages/fastify-renderer/test/FastifyRenderer.spec.ts

```diff
@@ -0,0 +1,164 @@
+  });
+
+  test('should decorate the fastify instance with a "setRenderConfig" method', async () => {
+    const server = await newFastify()
+    await server.register(FastifyRenderer)
+    expect(server.setRenderConfig).toBeInstanceOf(Function)
+  });
+
+  test('should set default render config for the instance', async () => {
+    const server = await newFastify()
+
+    expect(server[kRenderOptions]).toBeUndefined()
+    await server.register(FastifyRenderer)
+    expect(server[kRenderOptions]).toBeInstanceOf(Object)
+  });
+
+  test('should set the renderOptions on the new fastify instance context', async () => {
+    const server = await newFastify()
+
+    await server.register(FastifyRenderer)
+    await server.register(async instance => {
+      expect(server[kRenderOptions]).toEqual(instance[kRenderOptions])
+    })
+  });
+
+  test('should mount vite routes at a prefix to avoid collision with user routes', async () => {
+    const server = await newFastify()
+
+    await server.register(FastifyRenderer)
+    expect(server.printRoutes()).toMatch('/.vite/')
+  });
+
+  // test.skip('should use config from vite dev server in dev mode', async () => {
+  // });
+
+  test('should close vite devServer when fastify server is closing in dev mode', async () => {
+    const devServer = await Vite.createServer();
+    const closeSpy = jest.spyOn(devServer, 'close')
+    jest.spyOn(Vite, 'createServer').mockImplementation(async () => devServer)
+
+    const server = await newFastify()
+    await server.register(FastifyRenderer, { devMode: true })
+    await server.listen(0)
+    await server.close()
+
+    expect(closeSpy).toHaveBeenCalled()
+  });
+
+  test('should prepare routes', async () => {
+    const server = await newFastify()
+    const prepareSpy = jest.spyOn(ReactRenderer.prototype, 'prepare')
+
+    await server.register(FastifyRenderer)
+    await server.listen(0)
+    await server.close()
+
+    expect(prepareSpy).toHaveBeenCalled()
+  });
+
+  test('should do nothing if the registered route is not renderable', async () => {
+    const server = await newFastify()
+    const registerRouteSpy = jest.spyOn(FastifyRendererPlugin.prototype, 'registerRoute')
+
+    await server.register(FastifyRenderer)
+    server.get('/', async (_request, reply) => reply.send('Hello'))
+    await server.inject({ method: 'GET', url: '/' })
+
+    expect(registerRouteSpy).toHaveBeenCalledTimes(0)
+  });
+
+  test('should register the route in the plugin if it\'s renderable', async () => {
+    const server = await newFastify()
+    const registerRouteSpy = jest.spyOn(FastifyRendererPlugin.prototype, 'registerRoute').mockImplementation(jest.fn())
+
+    await server.register(FastifyRenderer)
+    server.get('/', { render: 'renderable-module-path' }, async (request, reply) => reply.send('Hello'))
+    await server.inject({ method: 'GET', url: '/' })
+
+    expect(registerRouteSpy).toHaveBeenCalledTimes(1)
+  });
```

Comment:
I'd say this test is again testing inner "private" implementation details as opposed to a nice interface such that this test is maintainable as the innards change. Any time you have to ninja in and mock something, you're creating a tiny little brittle bit that makes the thing harder to change, because now you have to change both the code itself *and* the tests that are ninja-ing in. Ideally, you could change the code to do something new in a different way, and the old tests would still pass without being changed. This requires artful selection of what you couple to in the tests. For more than you might ever want on the subject you can look at https://martinfowler.com/articles/mocksArentStubs.html